### PR TITLE
Organize Trainer flags

### PR DESCRIPTION
**Pull request by kaushikb11**
## What does this PR do?

Was hard to see the last three flags breaking the alpha order ☝🏽

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [ ] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [ ] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (if necessary)
- [ ] Did you write any new necessary tests? (not for typos and docs)
- [ ] Did you verify new and existing tests pass locally with your changes?
- [ ] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

File: pytorch_lightning/trainer/trainer.py [hunk 1]
```diff
@@ -258,6 +258,14 @@ def __init__(
                 :class:`datetime.timedelta`, or a dictionary with keys that will be passed to
                 :class:`datetime.timedelta`.
 
+            move_metrics_to_cpu: Whether to force internal logged metrics to be moved to cpu.
```
Comment (awaelchli) [inline pytorch_lightning/trainer/trainer.py:261]: there are other items that are not alphabetically ordered, e.g., the min/max steps and overfit_batches as far as I can tell

Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py:261]: yeah, I don't think we shouldn't change min/max steps (user readability and convenience).

Comment (kaushikb11) [reply pytorch_lightning/trainer/trainer.py:243]: Actually, there were more 😂 😂
